python/cuml/cuml/dask/__init__.py

```diff
@@ -12,7 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from dask import config
+try:
+    import dask
+    import dask_cudf as _  # noqa
+    import raft_dask as _  # noqa
+    import dask.distributed as _  # noqa
+
+    del _
+except ImportError as exc:
+    msg = (
+        f"{exc!s}\n\n"
+        "CuML Dask requirements are not installed.\n\n"
+        "Please either conda or pip install as follows:\n\n"
+        "  conda install cuml-dask      # either conda install\n"
+        '  pip install -U "cuml[dask]"  # or pip install\n'
+    )
+    raise ImportError(msg) from exc
```

Comment:
Here we raise a nicer `ImportError` when a user tries to import something from `cuml.dask` when all dependencies aren't installed.